File: clojure/prime-factors/src/prime_factors.clj
(ns prime-factors)

(defn of
  "Returns the prime factors of the given number."
  [num]
  (loop [primes [] candidate 2 remaining num]
    (cond
      (= 1 remaining) primes
      (zero? (mod remaining candidate)) (recur 
                                         (conj primes candidate) 
                                         candidate 
                                         (quot remaining candidate))
      :else (recur 
             primes 
             (if (= 2 candidate) 3 (+ 2 candidate)) 
             remaining))))

File: python/prime-factors/prime_factors.py
"""Return all prime factors of a given value"""

def factors(value):
    prime_factors = []
    candidate = 2
    remaining = value

    while remaining > 1:
        if remaining % candidate == 0:
            prime_factors.append(candidate)
            remaining //= candidate
        else:
            candidate = 3 if candidate == 2 else candidate + 2

    return prime_factors
    